cmd/snap/cmd_buy.go

```diff
@@ -71,6 +71,10 @@ func (x *cmdBuy) Execute(args []string) error {
 func buySnap(opts *store.BuyOptions) error {
 	cli := Client()
 
+	if !cli.LoggedIn() {
+		return fmt.Errorf(i18n.G("You need to be logged in to purchase software. Please run ‘snap login’ and try again."))
```

Comment:
Apologies. Clearly should have been more careful copy and pasting from the UX doc!